packages/client/src/internals/GenericTRPCClient.ts

```diff
@@ -48,17 +44,17 @@ export interface TRPCSubscriptionObserver<TValue, TError> {
 }
 
 /** @internal */
-export type CreateTRPCClientOptions<TRouter extends AnyRouter> =
+export type CreateTRPCClientOptions<TRouter extends AnyRouter = AnyRouter> =
   | CreateTRPCClientBaseOptions & {
       links: TRPCLink<TRouter>[];
     };
 
-export class TRPCClient<TRouter extends AnyRouter> {
-  private readonly links: OperationLink<TRouter>[];
+export class GenericTRPCClient {
```

Comment:
Does this need to be a `class`? 🤔 

I had an issue somewhere to rewrite everything to not use `class` b/c `class`es suck in JS

Reply:
Can we avoid exporting this as a `class`?

I'm fine if we export it as a `type` but if we export the class it means someone somewhere will do `new GenericTRPCClient`

Let's only export the factory method and it's interface so we're free to refactor it class-free

Reply:
Am I missing something or is this not exactly how it works currently?